Frontend/CO.CDP.OrganisationApp/Pages/Registration/OneLoginCallback.cshtml.cs

```diff
@@ -16,28 +16,34 @@ public async Task<IActionResult> OnGet()
         var userInfo = await oneLoginClient.GetUserInfo()
                     ?? throw new Exception("Unable to retrive user info"); // show error page?
 
+        Tenant.WebApiClient.Tenant? tenant;
+
         try
         {
-            var tenant = await tenantClient.CreateTenantAsync(
-                new NewTenant(
-                    new TenantContactInfo(userInfo.Email, userInfo.PhoneNumber),
-                    userInfo.UserId));
-
-            session.Set(Session.RegistrationDetailsKey,
-            new RegistrationDetails
-            {
-                TenantId = tenant.Id,
-                FirstName = tenant.Name,
-                Email = tenant.ContactInfo.Email,
-            });
-
-            return RedirectToPage("YourDetails");
+            tenant = await tenantClient.LookupTenantAsync(userInfo.UserId);
         }
-        catch (ApiException)
+        catch (ApiException ex) when (ex.StatusCode == 404)
         {
-            // show error page?
-            throw new Exception("Unable to create tenant");
+            try
+            {
+                tenant = await tenantClient.CreateTenantAsync(
```

Comment:
I think we should first query the tenant before we attempt to create it. Querying for a tenant is not an exceptional path. Furthermore, people will log in far more often than they register. Trying to register them every time triggers unnecessary calls.

Reply:
We are doing tenant lookup already here @jakzal, are you suggesting doing a get also?

Reply:
Ignore me. Looking at the diff it looked like the order was reversed (create, then lookup). Now that I just opened that file it's ok!